src/brainfuck/ArgParser.java

```diff
@@ -27,6 +27,9 @@ public ArgParser(String[] args) throws SyntaxException {
 				case "-p":
 					if (i+1 < args.length && !(args[i+1].startsWith("-"))) {
 						this.filename = args[i+1];
+						if (this.filename.endsWith(".bmp")) {
```

Comment:
Will probably needs to be modified to check MIME type (or BMP header) rather than the extension. Also it should be case insensitive (ie. match ".BMP" too as they like to use it on older Windows versions).